controllers/clustergroupupgrade_controller.go

```diff
@@ -247,296 +310,227 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
-										&clusterGroupUpgrade.Status.Conditions,
-										utils.ConditionTypes.Succeeded,
-										utils.ConditionReasons.UpgradeTimedOut,
-										metav1.ConditionFalse,
-										"The ClusterGroupUpgrade CR policies are taking too long to complete (Aborting on failure)",
-									)
-								default:
-									// If the value was continue or not defined then continue
-									clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt = metav1.Time{}
-									if clusterGroupUpgrade.Status.Status.CurrentBatch < len(clusterGroupUpgrade.Status.RemediationPlan) {
-										clusterGroupUpgrade.Status.Status.CurrentBatch++
-									}
+							default:
+								// If the value was continue or not defined then continue
+								clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt = metav1.Time{}
+								if clusterGroupUpgrade.Status.Status.CurrentBatch < len(clusterGroupUpgrade.Status.RemediationPlan) {
+									clusterGroupUpgrade.Status.Status.CurrentBatch++
 								}
 							}
 						}
 					}
 				}
+			}
+		} else {
+			// On last batch, check all batches
+			var isUpgradeComplete bool
+			isUpgradeComplete, err = r.isUpgradeComplete(ctx, clusterGroupUpgrade)
+			if err != nil {
+				return
+			}
+			if isUpgradeComplete {
+				utils.SetStatusCondition(
+					&clusterGroupUpgrade.Status.Conditions,
+					utils.ConditionTypes.Progressing,
+					utils.ConditionReasons.Completed,
+					metav1.ConditionFalse,
+					"The ClusterGroupUpgrade CR has all clusters compliant with all the managed policies",
+				)
+				utils.SetStatusCondition(
+					&clusterGroupUpgrade.Status.Conditions,
+					utils.ConditionTypes.Succeeded,
+					utils.ConditionReasons.UpgradeCompleted,
+					metav1.ConditionTrue,
+					"The ClusterGroupUpgrade CR has all clusters compliant with all the managed policies",
+				)
+				nextReconcile = requeueImmediately()
 			} else {
-				// On last batch, check all batches
-				var isUpgradeComplete bool
-				isUpgradeComplete, err = r.isUpgradeComplete(ctx, clusterGroupUpgrade)
+				err = r.remediateCurrentBatch(ctx, clusterGroupUpgrade, &nextReconcile)
 				if err != nil {
 					return
 				}
-				if isUpgradeComplete {
-					utils.SetStatusCondition(
-						&clusterGroupUpgrade.Status.Conditions,
-						utils.ConditionTypes.Progressing,
-						utils.ConditionReasons.Completed,
-						metav1.ConditionFalse,
-						"The ClusterGroupUpgrade CR has all clusters compliant with all the managed policies",
-					)
-					utils.SetStatusCondition(
-						&clusterGroupUpgrade.Status.Conditions,
-						utils.ConditionTypes.Succeeded,
-						utils.ConditionReasons.UpgradeCompleted,
-						metav1.ConditionTrue,
-						"The ClusterGroupUpgrade CR has all clusters compliant with all the managed policies",
-					)
-					nextReconcile = requeueImmediately()
-				} else {
-					err = r.remediateCurrentBatch(ctx, clusterGroupUpgrade, &nextReconcile)
-					if err != nil {
-						return
-					}
-				}
 			}
 		}
-
 	}
 
 	// Update status
```

Comment:
We need to handle the error here.

Reply:
Error from where?

Reply:
Sorry, the comment is not related to this PR, so i couldn't add the comment in line 543. It is in line 543:

	err = r.updateStatus(ctx, clusterGroupUpgrade)
	return

Reply:
Ok. The err is returned. It's not an error we should be handling.

Reply:
you are right.